packages/react-components/src/hooks/useAlarms/utils/alarmDataUtils.ts

```diff
@@ -0,0 +1,141 @@
+import {
+  AssetProperty,
+  DescribeAssetModelResponse,
+  DescribeAssetResponse,
+} from '@aws-sdk/client-iotsitewise';
+import { AlarmData, AlarmDataStatus, AlarmRequest } from '../types';
+import {
+  buildFromCompositeModel,
+  isAlarmCompositeModel,
+} from './compositeModelUtils';
+
+/**
+ * This function builds an AlarmData object for each alarm composite model on a SiteWise asset.
+ * The content of AlarmData depends on the AlarmRequest argument.
+ *
+ * @param request is an AlarmRequest that may have an assetId
+ * @param status is the status of the DescribeAssetResponse query
+ * @param assetResponse is the DescribeAssetResponse
+ * @returns a list of AlarmData for all alarms on an asset
+ */
+export const buildFromAssetResponse = ({
+  request,
+  status,
+  assetResponse,
+}: {
+  request: AlarmRequest;
+  status: AlarmDataStatus;
+  assetResponse?: DescribeAssetResponse;
+}): AlarmData[] => {
+  const { assetId, assetCompositeModelId, inputPropertyId } = request;
+
+  if (!assetResponse || !assetId) {
+    return [];
+  }
+
+  if (assetResponse && assetId && assetResponse?.assetId !== assetId) {
+    throw new Error(
+      'AlarmDataFactory.buildFromAssetResponse expects there to be an assetId in the request that matches the DescribeAssetResponse'
+    );
```

Comment:
Who is this error intended for?

Reply:
This is intended for a developer who uses this utility. I see this may be unnecessary. I can instead log this error and return a default response.